projects/ipdg_stokes/post_processing/mesh_function_interpolation.h

```diff
@@ -0,0 +1,105 @@
+   * @param from The refinement level of the coarser mesh
+   * @param to The ferinement level of the finer mesh
+   *
+   * Currently only meshes with affine geometries are supported. This could
+   * change if a function lf::geometry::Geometry::Local(Eigen::MatrixXd &global)
+   * would be provided.
+   */
+  MeshFunctionInterpolation(const INNER_MF &inner,
+                            const lf::refinement::MeshHierarchy &meshes,
+                            lf::base::size_type from, lf::base::size_type to)
+      : inner_(inner),
+        parents_(meshes.getMesh(to)->NumEntities(0)),
+        mesh_to_(meshes.getMesh(to)) {
+    // Initialize a map from entities in the fine mesh to entities in the coarse
+    // mesh
+    for (const auto cell : mesh_to_->Entities(0)) {
+      parents_[mesh_to_->Index(*cell)] = parent(cell, meshes, from, to);
+    }
+  }
+
+  decltype(auto) operator()(const lf::mesh::Entity &entity,
+                            const Eigen::MatrixXd &local) const {
+    // Get the element in the coarser mesh which is the parent of entity
+    const lf::mesh::Entity *parent = parents_[mesh_to_->Index(entity)];
+    // Map the evaluation points from local coordinates in entity to local
+    // coordinates in parent HACK: This assumes that the zeroth vertex of the
+    // entity has local coordinates (0,0)
+    const auto entity_geom = entity.Geometry();
+    const auto parent_geom = parent->Geometry();
+    if (!parent_geom->isAffine()) {
+      throw lf::base::LfException(
+          "MeshFunctionInterpolation only works with affine elements");
+    }
+    const Eigen::MatrixXd global = entity_geom->Global(local);
+    const Eigen::MatrixXd global_shifted =
+        global -
+        parent_geom->Global(Eigen::MatrixXd::Zero(local.rows(), local.cols()));
+    const Eigen::MatrixXd parent_local =
+        parent_geom->JacobianInverseGramian(Eigen::Vector2d::Zero()) *
+        global_shifted;
+    // Evaluate the inner mesh function at the local coordinate inside parent
+    return inner_(*parent, parent_local);
+  }
+
+ private:
+  const INNER_MF inner_;
+  std::vector<const lf::mesh::Entity *> parents_;
+  std::shared_ptr<const lf::mesh::Mesh> mesh_to_;
+
+  /**
+   * @brief Find the parent entity given a child entity
+   * @param entity A pointer to the cild entity
+   * @param meshes The MeshHierarchy in which to find the parent
+   * @param from The refinement level of the parent
+   * @param to The refinement level of the cild entity
+   */
+  static const lf::mesh::Entity *parent(
+      const lf::mesh::Entity *entity,
+      const lf::refinement::MeshHierarchy &meshes, lf::base::size_type from,
+      lf::base::size_type to) {
+    if (from == to) {
+      return entity;
+    } else {
+      const auto mesh_to = meshes.getMesh(to);
+      const lf::mesh::Entity *immediate_parent =
+          meshes.ParentInfos(to, 0)[mesh_to->Index(*entity)].parent_ptr;
+      return parent(immediate_parent, meshes, from, to - 1);
```

Comment:
Just a tip: recursive function calls introduce a lot of overhead (unless the compiler can optimize it out). I don't think it matters much here, but next time you can maybe use a loop instead ;)